### DISCUSSION TITLE
Data Storage sended by an app

### DISCUSSION THREAD

**Discussion by antiadominguez** [edgexfoundry/community#296] [Q&A] [answered]
I am searching for a way to storage on core-data the data sended by an app-service. I have an app-service that analyze the events of a device and post the result back to the bus. I found that core-data only listen to event topic and I would like to know if exists the way to listen to application service topic.

**Comment by jiekechoo** [comment]
@antiadominguez 
We need more information about your app-service.
There are some tips for you:
- Using the app-service sends events to message-bus in formated topics: [PublishWithTopic](https://github.com/edgexfoundry/app-functions-sdk-go/blob/main/internal/app/service.go#L766)
- By default you could use the rule-engine(eKuiper) analyzing the events

Jieke Choo

**Reply by antiadominguez** [reply]
My app-service has an IA module inside, so I want to catch device events, pass them through my IA and then publish the result back to the bus and storage on the core-data if it is possible. I think eKuiper is not capable to do that, so I program my own app-service. I will try PublishWithTopic as you commented but here you have more info about my app

**Comment by cloudxxx8** [comment]
@antiadominguez if you are writing your own app service, you can leverage the event client to send data to Core Data with REST API
https://docs.edgexfoundry.org/3.1/microservices/application/sdk/api/ApplicationServiceAPI/#eventclient

**Reply by antiadominguez** [reply]
I don't understand what you mean with that, if you can explain it in detail

**Reply by jiekechoo** [reply, marked as answer]
There are a few steps that you have to do in your app-service.
- Setup `Core-Data` client in the configuration.yaml file
- Using `EventClient` API to create a client as @cloudxxx8 mentioned
- Send events to the message-bus

**Reply by antiadominguez** [reply]
I'm not understanding what you mean with the second step, I don't know how to use EventClient API and the documentation is very confusing

**Reply by cloudxxx8** [reply]
It's the Golang client code, so the first question is whether you are writing your own app service.
You can use HTTP export or MQTT export to achieve the same thing, if you are using app configurable.

**Reply by antiadominguez** [reply]
Yes, I am writing my own app service starting on the template
